File: Java/CoinApi/app/src/main/java/com/example/ageapipractice/MainActivity.java
package com.example.ageapipractice;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.text.Editable;
import android.view.View;
import android.widget.Toast;

import com.example.ageapipractice.databinding.ActivityMainBinding;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class MainActivity extends AppCompatActivity {

    private Editable coinid;

    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        coinid = binding.editText.getText();

        GetAPI getAPI = ApiProvider.getRetrofit().create(GetAPI.class);

        binding.Button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                getAPI.getGET(coinid).enqueue(new Callback<GetResponse>() {
                    @Override
                    public void onResponse(Call<GetResponse> call, Response<GetResponse> response) {
                        if(response.isSuccessful()){
                            binding.Id.setText(response.body().getId());
                            binding.Name.setText(response.body().getName());
                            binding.Symbol.setText(response.body().getSymbol());
                            binding.Rank.setText(response.body().getRank());
                        }else{
                            Toast.makeText(MainActivity.this, "Error!", Toast.LENGTH_SHORT).show();
                        }
                    }

                    @Override
                    public void onFailure(Call<GetResponse> call, Throwable t) {
                        Toast.makeText(MainActivity.this, "통신 실패", Toast.LENGTH_SHORT).show();
                    }
                });
            }
        });
    }
}

File: Kotlin/MovieApi/app/src/main/java/com/example/movieapi/MainActivity.kt
package com.example.movieapi

import android.icu.text.CaseMap
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.text.Editable
import android.widget.EditText
import android.widget.Toast
import com.example.movieapi.databinding.ActivityMainBinding
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    private lateinit var title : Editable

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        title = binding.etInput.getText()

        binding.btSearch.setOnClickListener {  RetrofitBuilder.MovieApi.getMovie(title).enqueue(object : Callback<Movie>{

            override fun onResponse(call: Call<Movie>, response: Response<Movie>) {
                if(response.isSuccessful){
                    val movieInfo = response.body()

                    binding.Title.text = movieInfo!!.Title
                    binding.Year.text = movieInfo!!.Year
                    binding.Runtime.text = movieInfo!!.Runtime
                    binding.Country.text = movieInfo!!.Country
                }else{
                    Toast.makeText(this@MainActivity, "Error!", Toast.LENGTH_SHORT).show()
                }
            }

            override fun onFailure(call: Call<Movie>, t: Throwable) {
                Toast.makeText(this@MainActivity, "통신실패", Toast.LENGTH_SHORT).show()
            }
        }) }


    }
}